### PR TITLE
ci: Rename GitHub workflows, add and unify names

### PR DESCRIPTION
**Pull request by janpio**
Clean up GH Action workflow names, file names and job names a bit.

Relies on https://github.com/prisma/engineer/pull/118

### CODE REVIEW

**Review by Jolg42 — APPROVED** [review]
Nice! ✨

File: .github/workflows/test-unit-tests.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Unit tests
+name: Test Unit tests
```
Comment (Jolg42) [inline .github/workflows/test-unit-tests.yml:1]: That sounds weird / repetitive?

Comment (janpio) [reply .github/workflows/test-unit-tests.yml:1]: Yes, technically "Test units" would be enough - but that sounded even weirder.
And "Run unit tests" would not fit the pattern that I am trying to have in all the job names 🤷 
Suggestions?

image

Comment (Jolg42) [reply .github/workflows/test-unit-tests.yml:1]: Well, what you suggested is fine I think

File: .buildkite/engineer [hunk 2]
```diff
@@ -54,7 +54,7 @@ fi
 # Check if the system has engineer installed, if not, use a local copy.
 if ! type "engineer" &> /dev/null; then
     # Setup Prisma engine build & test tool (engineer).
-    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/1.64/latest/$OS/engineer.gz" --output engineer.gz
+    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/janpio-patch-1/latest/$OS/engineer.gz" --output engineer.gz
```
Comment (janpio) [inline .buildkite/engineer:57]: To be updated after release of changes from https://github.com/prisma/engineer/pull/118

Comment (janpio) [reply .buildkite/engineer:57]:
```suggestion
    curl --fail -sSL "https://prisma-engineer.s3-eu-west-1.amazonaws.com/1.65/latest/$OS/engineer.gz" --output engineer.gz
```

**Pull request merged**